tests/test_worker_timeout.py

```diff
@@ -100,8 +101,10 @@ def test_worker_timeout_with_slowly_pickling_objects(self, n_tasks=5):
                     id, [SlowlyPickling(delay)] * n_tasks))
                 assert len(results) == n_tasks
 
+    @pytest.mark.skipif(hasattr(sys, "pypy_version_info"),
+                        reason="Warning detection are unreliable on pypy")
```

Comment:
Ideally it would be great to find a way to only skip the warning check but still run the tests in pypy to ensure that there is no deadlock.

For instance catch and record all warnings and then do a manual assert at the end of the test to ensure that if you are not under PyPy there is a UserWarning with the right message an nothing else.